system-contract-dapp-playground/src/components/contract-interaction/erc/erc-20/methods/TokenInformation.tsx

```diff
@@ -18,48 +18,195 @@
  *
  */
 
+import { useState } from 'react';
 import { Contract } from 'ethers';
+import { useToast } from '@chakra-ui/react';
 import OneLineMethod from '../common/OneLineMethod';
+import { CommonErrorToast } from '@/components/toast/CommonToast';
+import { getERC20TokenInformation } from '@/api/hedera/erc20-interactions';
 
 interface PageProps {
   baseContract: Contract;
 }
 
 const TokenInformation = ({ baseContract }: PageProps) => {
+  const toaster = useToast();
+  const [tokenInfo, setTokenInfo] = useState({
+    name: {
+      result: '',
+      isLoading: false,
+    },
+    symbol: {
+      result: '',
+      isLoading: false,
+    },
+    totalSupply: {
+      result: '',
+      isLoading: false,
+    },
+    decimals: {
+      result: '',
+      isLoading: false,
+    },
+  });
+
+  /**
+   * @dev handle execute token information queries
+   */
+  const handleExecutingMethods = async (method: 'name' | 'symbol' | 'totalSupply' | 'decimals') => {
+    let errTitle;
+
+    switch (method) {
+      case 'name':
```

Comment:
Move the shared logic between all the switch cases into a shared method.
Should be able to craft one that take in the token info label